wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Comment/Edit.ts

```diff
@@ -0,0 +1,329 @@
+/**
+ * Provides editing support for comments.
+ *
+ * @author  Alexander Ebert
+ * @copyright  2001-2019 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Comment/Edit
+ */
+
+import * as Ajax from "../../Ajax";
+import { AjaxCallbackSetup, ResponseData } from "../../Ajax/Data";
+import * as Core from "../../Core";
+import DomChangeListener from "../../Dom/Change/Listener";
+import DomUtil from "../../Dom/Util";
+import * as Environment from "../../Environment";
+import * as EventHandler from "../../Event/Handler";
+import * as Language from "../../Language";
+import { RedactorEditor } from "../Redactor/Editor";
+import * as UiScroll from "../Scroll";
+import * as UiNotification from "../Notification";
+
+interface AjaxResponse {
+  actionName: string;
+  returnValues: {
+    message: string;
+    template: string;
+  };
+}
+
+class UiCommentEdit {
+  protected _activeElement: HTMLElement | null = null;
+  protected readonly _comments = new Set<HTMLElement>();
+  protected readonly _container: HTMLElement;
+  protected _editorContainer: HTMLElement | null = null;
+
+  /**
+   * Initializes the comment edit manager.
+   */
+  constructor(container: HTMLElement) {
```

Comment:
```suggestion
  constructor(private readonly _container: HTMLElement) {
```
(even though it's probably inconsistent by now)

Reply:
It is redundant, because `this._container` is set in the parent constructor. Also I try to avoid mixing constructor property promotion with regular properties.